### PR TITLE
chore: Refine native dashboard cleanup logic

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Making a few minor refinements on the native dashboard filter cleanup logic.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/cli/native_filters.py [hunk 1]
```diff
@@ -368,17 +368,15 @@ def cleanup(
             json_metadata = json.loads(dashboard.json_metadata or "{}")
             position_json = json.loads(dashboard.position_json or "{}")
 
-            if "native_filter_migration" not in json_metadata:
```
Comment (john-bodley) [inline superset/cli/native_filters.py:371]: If a dashboard has no filters then this key wouldn't exit and thus it's a misnomer to say that the dashboard hadn't been upgraded.

File: superset/cli/native_filters.py [hunk 2]
```diff
@@ -368,17 +368,15 @@ def cleanup(
...
-            del json_metadata["native_filter_migration"]
-            dashboard.json_metadata = json.dumps(json_metadata)
+            if "native_filter_migration" in json_metadata:
+                del json_metadata["native_filter_migration"]
+                dashboard.json_metadata = json.dumps(json_metadata)
 
             for value in position_json.values():
                 if (
                     isinstance(value, dict)
+                    and value["type"] == "MARKDOWN"
```
Comment (john-bodley) [inline superset/cli/native_filters.py:379]: Not all values in the the `position_json` dictionary contain the `meta` key.

Comment (michael-s-molina) [inline superset/cli/native_filters.py:379]:
```suggestion
                    and "meta" in value
```

Comment (john-bodley) [reply superset/cli/native_filters.py:381]: @michael-s-molina I've updated the PR to use a variant of this (in the form of the walrus operator) throughout for consistency.

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**